### PR TITLE
implementation of the polyfill detailed page

### PR DESCRIPTION
**Pull request by reiniergs**

### CODE REVIEW

File: src/main/java/org/polyfill/components/Polyfill.java [hunk 1]
```diff
@@ -135,6 +137,35 @@ public int hashCode() {
         return hash;
     }
 
+    /**
+     * Gets the type of license of the polyfill implementation e.g. MIT | Apache
+     * @return {String} - The type of license
+     */
+    public String getLicense() {
```
Comment (scottmo) [inline src/main/java/org/polyfill/components/Polyfill.java]: return getStringFromMap(this.polyfillMap, LICENSE);

File: src/main/java/org/polyfill/components/Polyfill.java [hunk 2]
```diff
@@ -135,6 +137,35 @@ public int hashCode() {
...
+        return (String) this.polyfillMap.get(LICENSE);
+    }
+
+    /**
+     * Gets the URL to the polyfill repository
+     * @return {String} - The URl to the repository
+     */
+    public String getRepository() {
```
Comment (scottmo) [inline src/main/java/org/polyfill/components/Polyfill.java]: return getStringFromMap(this.polyfillMap, REPO);

File: src/main/java/org/polyfill/services/PreSortPolyfillQueryService.java [hunk 3]
```diff
@@ -209,4 +209,8 @@ private boolean shouldIncludeFeature(FeatureOptions featureOptions, UserAgent us
     public Map<String, Polyfill> getAllPolyfills() {
         return polyfills;
     }
+
+    public Polyfill getPolyfillByName(String name) {
```
Comment (scottmo) [inline src/main/java/org/polyfill/services/PreSortPolyfillQueryService.java]: add to interface as well

File: src/main/java/org/polyfill/views/JsonView.java [hunk 4]
```diff
@@ -16,18 +16,28 @@
...
+
     public String getContentType() {
         return MediaType.APPLICATION_JSON_UTF8_VALUE;
     }
 
     public void render(Map<String, ?> map, HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse) throws Exception {
         httpServletResponse.setContentType(getContentType());
 
+        if (statusCode != null) {
+           httpServletResponse.setStatus(this.statusCode);
```
Comment (scottmo) [inline src/main/java/org/polyfill/views/JsonView.java]: 3 spaces!!!! change to 4 spaces

File: src/main/java/org/polyfill/components/Polyfill.java [hunk 5]
```diff
@@ -135,6 +137,35 @@ public int hashCode() {
...
+        return (String) this.polyfillMap.get(REPO);
+    }
+
+    public Map<String, Object> toMap() {
```
Comment (scottmo) [inline src/main/java/org/polyfill/components/Polyfill.java]: I think we should have this somewhere else. Polyfill should stay simple and should not contain knowledge about what to return in toMap. The most we can do is to return this.polyfills.clone() and let remove the unnecessary fields somewhere else.

File: src/main/java/org/polyfill/components/Polyfill.java (hunk 1 above)
Comment (reiniergs) [inline src/main/java/org/polyfill/components/Polyfill.java]: done here

File: src/main/java/org/polyfill/components/Polyfill.java (hunk 2 above)
Comment (reiniergs) [inline src/main/java/org/polyfill/components/Polyfill.java]: done here

**Comment by reiniergs** [comment]
@scottmo let's open an issue for move toMap implementation out of Polyfill, I'm agree

**Pull request merged**